### PR TITLE
chore(renovate): separate test and dev deps, disable backstage updates

### PR DESCRIPTION
**Pull request by kim-tsao**
## Description

Please explain the changes you made here.

## Which issue(s) does this PR fix
Trying to reduce the noise and break up the PRs into smaller changes
- Fixes #?
 https://issues.redhat.com/browse/RHIDP-2883
## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

### CODE REVIEW

File: .github/renovate.json [hunk 1]
```diff
@@ -49,6 +62,11 @@
       ],
       "groupName": ["Backstage packages"],
       "dependencyDashboardApproval": true
+    },
+    {
+      "matchDepPatterns": ["^@backstage/"],
```
Comment (nickboldt) [inline .github/renovate.json:67]: will this affect the plugins we're moving from janus-idp/backstage-plugins to backstage/community-plugins ?

File: .github/renovate.json [hunk 2]
```diff
@@ -38,9 +40,20 @@
...
+      "extends": ["packages:test"],
+      "matchUpdateTypes": ["patch", "minor"],
+      "matchCurrentVersion": "!/^0/",
+      "groupName": "Test packages (non-major)",
+      "automerge": true
+    },
+    {
+      "extends": ["packages:test"],
+      "matchUpdateTypes": ["major"],
+      "groupName": "Test packages"
```
Comment (nickboldt) [inline .github/renovate.json:53]:
```suggestion
      "groupName": "Test packages (major)"
```

File: .github/renovate.json [hunk 3]
```diff
@@ -38,9 +40,20 @@
       "groupName": ["DevDependencies (non-major)"],
       "automerge": true
     },
+    {
+      "extends": ["packages:test"],
+      "matchUpdateTypes": ["patch", "minor"],
+      "matchCurrentVersion": "!/^0/",
+      "groupName": "Test packages (non-major)",
```
Comment (nickboldt) [inline .github/renovate.json:47]:
```suggestion
      "groupName": "Test packages (minor and patch)",
```

File: .github/renovate.json (hunk 2 above)
Comment (kim-tsao) [reply .github/renovate.json:53]: don't need to add that.  Renovate will do it for you. See https://github.com/janus-idp/backstage-plugins/blob/main/.github/renovate.json#L93-L99 and test PRs in dependency dashboard: https://github.com/janus-idp/backstage-plugins/issues/19

File: .github/renovate.json (hunk 1 above)
Comment (kim-tsao) [reply .github/renovate.json:67]: yeah it does.  I don't know what the release decision for those upstream plugins will be.  Last I heard, they were going to stay a release behind.

File: .github/renovate.json (hunk 3 above)
Comment (kim-tsao) [reply .github/renovate.json:47]: same diff, just tried to keep it consistent with all the other rules and naming convention unless you want me to change those too

**Review by schultzp2020 — APPROVED** [review]
/lgtm
/approve

File: .github/renovate.json (hunk 1 above)
Comment (kim-tsao) [reply .github/renovate.json:67]: @schultzp2020 confirmed this has no impact as the community plugins have a different ns `@backstage-community/*`

**Pull request merged**